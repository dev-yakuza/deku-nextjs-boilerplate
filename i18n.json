{
  "locales": ["ja"],
  "defaultLocale": "ja",
  "localeDetection": false,
  "pages": {
    "*": ["common"]
  }
}
