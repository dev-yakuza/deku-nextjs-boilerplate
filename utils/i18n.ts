import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import { i18n as i18nConfig } from '../next-i18next.config'

i18n.use(initReactI18next).init({
  fallbackLng: 'en',
  resources: Object.assign(
    {},
    ...i18nConfig.locales.map((locale) => ({
      [locale]: {
        common: require(`../public/locales/${locale}/common.json`),
      },
    })),
  ),
})

export default i18n
