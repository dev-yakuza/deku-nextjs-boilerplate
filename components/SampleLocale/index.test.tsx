import { useEffect } from 'react'

import { render, screen } from '@testing-library/react'
import { I18nextProvider } from 'react-i18next'

import i18n from 'utils/i18n'

import { SampleLocale } from './index'

interface Props {
  readonly locale: 'en' | 'ko' | 'ja'
}
const TestComponent = ({ locale }: Props) => {
  useEffect(() => {
    i18n.changeLanguage(locale)
  }, [locale])

  return (
    <I18nextProvider i18n={i18n}>
      <SampleLocale />
    </I18nextProvider>
  )
}

describe('<SampleLocale />', () => {
  it('displayed well', () => {
    const { container, rerender } = render(<TestComponent locale="en" />)

    expect(
      screen.queryByText('English | Korean | Japanese'),
    ).toBeInTheDocument()
    expect(screen.queryByText('영어 | 한국어 | 일본어')).not.toBeInTheDocument()
    expect(screen.queryByText('英語 | 韓国語 | 日本語')).not.toBeInTheDocument()

    expect(container).toMatchSnapshot()

    rerender(<TestComponent locale="ko" />)
    expect(
      screen.queryByText('English | Korean | Japanese'),
    ).not.toBeInTheDocument()
    expect(screen.queryByText('영어 | 한국어 | 일본어')).toBeInTheDocument()
    expect(screen.queryByText('英語 | 韓国語 | 日本語')).not.toBeInTheDocument()

    rerender(<TestComponent locale="ja" />)
    expect(
      screen.queryByText('English | Korean | Japanese'),
    ).not.toBeInTheDocument()
    expect(screen.queryByText('영어 | 한국어 | 일본어')).not.toBeInTheDocument()
    expect(screen.queryByText('英語 | 韓国語 | 日本語')).toBeInTheDocument()
  })
})
