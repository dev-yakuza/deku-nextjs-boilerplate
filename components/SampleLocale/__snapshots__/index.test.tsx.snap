// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleLocale /> displayed well 1`] = `
<div>
  <p>
    English
     | 
    Korean
     | 
    Japanese
  </p>
</div>
`;
